lib/src/Base/Stat/openturns/Sample.hxx

```diff
@@ -319,10 +319,11 @@ public:
   Sample & operator /= (const SquareMatrix & scaling);
 
   /** Ranked sample */
-  Sample rank() const;
+  Sample rank(const Bool normalize = false) const;
 
   /** Ranked component */
-  Sample rank(const UnsignedInteger index) const;
+  Sample rank(const UnsignedInteger index,
+	      const Bool normalize = false) const;
```

Comment:
The signatures of rank are ambiguous here, what's `rank(0)` ? `rank(normallize=False)` or `rank(index=0)` ?

Reply:
Maybe add another rankNormalized signature ?

Reply:
This normalize flag looks useless to me, one can write `(sample+1.0)/sample.getSize()` if needed.

Reply:
No, you won't notice a difference, it will be much faster than rank() anyway.
BTW in EmpiricalBernsteinCopula::setCopulaSample, Sample copies can be avoided by writing
```
  if (isEmpiricalCopulaSample && remainder == 0)
    copulaSample_ = copulaSample;
  else
  {
    if (remainder == 0)
      copulaSample_ = copulaSample.rank();
    else
    {
      LOGINFO(OSS() << "Must drop the last " << remainder << " to build a EmpiricalBernsteinCopula as the given sample has a size=" << size << " which is not a multiple of the bin number=" << binNumber_);
      copulaSample_ = Sample(copulaSample, 0, size - remainder).rank();
    }
    // Normalize
    copulaSample_ += 1.0;
    copulaSample_ /= 1.0 * (size - remainder);
```

Or use 2 for loops to normalize if you want.

Reply:
But rank(index) could be dropped too, yes.

Reply:
I introduced the toEmpiricalCopula() method because it is a very common task to compare the normalized ranks of a sample and the iso-pdf of a copula learnt on this sample. Even if Denis's proposal is quite short, I am not sure that it is as helpful as a dedicated method for the standard user. For now I will remove the flag and normalize things directly into EmpiricalBernsteinCopula using Denis's code. I don't remember well the procedure to deprecate a method, it will be in another PR.